Fixes: #517

aashish-ak opened the pull request:
---

**What's this PR do?**
  It fixes #517 ,

**Any background context you want to provide?**
 The error 'page_params is not defined' occured because when the third party authorization windows 
 are opened at the time of login, so i added a check that if the variable `page_params` exist only then load the DOM.

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

aashish-ak committed:
preload: use page_params only when it exists.

This commits puts a try..catch to handle when page_params doesn't exists when the webpage is
not a zulip webapp like auth pages.
Fixes: #517

abhigyank commented:
Hello @aashish-ak thanks for doing this!
However I think the correct approach could be to use a `try..catch` for the `serverLanguage` variable. I don't think hard-coding the websites is a good solution.

aashish-ak commented:
Okay will it be better if i fix the hostname to `*.zulip.*` or `*.zulipchat.*` ?? Will it work?? @abhigyank

abhigyank commented:
No we can't do that, as many people host Zulip on-premise and they have different domains.
Can you try something like?
```js
let serverLanguage = null;
try{
    serverLanguage = page_params.default_language;
} 
```

aashish-ak commented:
@abhigyank this isn't helping as the the whole code in that block should be loaded if the window is not of a third party authorisation method.

aashish-ak commented:
@abhigyank I've modified the code, merge it if you find the PR worthy :smile:

abhigyank commented on app/renderer/js/notification/index.js:
```diff
@@ -20,8 +20,14 @@ if (process.platform === 'darwin') {
+		webpageParams = page_params;
+	} catch (err) {
+		webpageParams = null;
+	}// eslint-disable-next-line no-undef, camelcase
```
Can you move the comment to the new line?

aashish-ak replied:
@abhigyank  Done :+1:

abhigyank commented:
@aashish-ak Looks good to me, left a comment about a small change.
Also can you squash both the commits into a single commit. You can look up `git rebase -i` for the same [here](https://zulip.readthedocs.io/en/latest/git/fixing-commits.html?highlight=squash#squashing-commits).
Also the can you change the commit message as per our guidelines [here](https://zulip.readthedocs.io/en/latest/contributing/version-control.html?highlight=commit%20message#commit-messages) . You can also look at other commit messages for references.

abhigyank reviewed (approved):
@akashnimare LGTM. Can be merged.

akashnimare commented:
@aashish-ak, @abhigyank  since we are using the same code, I think we can wrap it up in a function and export it wherever we need, something like this -

```js
const isPageParams = () => {
    let webpageParams = null;
    try {
        webpageParams = page_params;
    } catch (err) {
        webpageParams = null;
    }
    return webpageParams;
}


module.exports = {
    isPageParams
}

```

abhigyank commented:
Right, this should probably go in `default-util` right?

akashnimare commented:
There or maybe we could create `params-util.js`.

akashnimare commented:
@aashish-ak are you interested in finishing this PR?

aashish-ak commented:
Sorry been a little busy, yes i'll finish it soon :smile:

aashish-ak commented:
I'll complete the work till noon, you can merge easily after that ☺️

akashnimare commented:
Awesome, would be great if you can finish it up since we're planning to add this in a new release.

aashish-ak committed:
preload: use page_params only when it exists.

This commits adds a params-utl.js file which has a function with a try..catch to handle when page_params doesn't exists when the webpage is
not a zulip webapp like auth pages.
Fixes: #517

aashish-ak commented:
@akashnimare @abhigyank It's done, sorry for the late commit.

akashnimare commented:
Merged after fixing linting errors and commit message. Thanks, @aashish-ak.
